packages/@vue/cli/lib/Generator.js

```diff
@@ -93,14 +95,45 @@ module.exports = class Generator {
     // exit messages
     this.exitLogs = []
 
+    this.pkg = Object.assign({
+      dependencies: {},
+      devDependencies: {}
+    }, pkg)
+
+    const pluginIds = plugins.map(p => p.id)
+
+    // load all the other plugins
+    this.otherPlugins = Object.keys(this.pkg.dependencies)
+      .concat(Object.keys(this.pkg.devDependencies))
+      .filter(isPlugin)
+      .filter(id => pluginIds.indexOf(id) === -1)
+
     const cliService = plugins.find(p => p.id === '@vue/cli-service')
     const rootOptions = cliService
       ? cliService.options
       : inferRootOptions(pkg)
+
+    // apply hooks from all plugins
+    this.otherPlugins.forEach(id => {
+      const api = new GeneratorAPI(id, this, {}, rootOptions)
+      const pluginGenerator = loadModule(`${id}/generator`, context)
+      if (pluginGenerator && pluginGenerator.hooks) {
+        pluginGenerator.hooks(api, {}, rootOptions, pluginIds)
```

Comment:
With this design & implementation, `afterInvoke` callbacks from other plugins also gets pushed into the callback arrays, this is not the expected behavior.
I think the `hooks` feature should be more restrictive. Object syntax may be a better alternative.

Reply:
As you can see here, I am resetting the non-any hooks. https://github.com/vuejs/vue-cli/pull/2337/files#diff-f0e2bfadbc7f0f35b5cc089613387802R126.

Reply:
Okay.